tools/adventure-pack/goodies/typescript/Iterator.prototype.toArray/index.test.ts

```diff
@@ -0,0 +1,25 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "./index";
+
+describe("Iterator.prototype.toArray", () => {
```

Comment:
I think there are a couple more tests that could be interesting:

1. Write a generator function that throws an error, what happens if we try to call `.toArray()` on the resulting iterator? I'm thinking it ought to throw.
2. Test on an iterator that doesn't come from an array, e.g. a Set's `.values()`.

...but these can come as a follow-up PR!